bspump/ftp/source.py

```diff
@@ -0,0 +1,159 @@
+import asab
+import asyncio
+import logging
+import asyncssh
+
+from ..abc.source import Source
+
+#
+
+L = logging.getLogger(__name__)
+
+#
+
+ConfigDefaults = {
+
+	'remote_path': '',
+	'local_path': '',
+
+}
+
+class FTPSource(Source):
+
+
+	def __init__(self, app, pipeline, connection, id=None, config=None):
+		super().__init__(app, pipeline, id=id, config=config)
+		self._connection = pipeline.locate_connection(app, connection)
+		self.Loop = app.Loop
+		self.App = app
+		self.Pipeline = pipeline
+
+		# self._connection.PubSub.subscribe("FTPConnection.open!", self._async_connection)
+		self.start(self.Loop) # without this start method, it wont run main(), I havent seen it in other sources (like amqp or mysql). Is this start method correct here?
+
+		self._rem_path = self.Config['remote_path']
+		self._loc_path = self.Config['local_path']
+		self._preserve = False, #True,
+		self._recurse = False, #True,
+
+
+	async def main(self):
+		await self._connection.ConnectionEvent.wait()
+		async with self._connection.acquire_connection() as connection:#._connection_check as connection:
+			async with connection.start_sftp_client() as sftp:
```

Comment:
@PremyslCerny 
Here this error appears: `asyncssh.misc.ChannelOpenError: Session request failed`